src/snowflake/cli/api/utils/models.py

```diff
@@ -12,3 +18,6 @@ def __getitem__(self, item):
 
     def __contains__(self, item):
         return item in os.environ or super().__contains__(item)
+
+    def update_from_dict(self, update_values: Dict[str, Any]):
```

Comment:
the original intention was to limit this to `UpdatableModel`. Should this inherit from it? Not quite sure how this change fits in the bigger picture.

Reply:
No. I tried to inherit from both UpdatableModel and dict, but they conflict. 
The way it's currently done, we use type Dict when parsing yml, which is recognized by Pydantic, and then transform to EnvironWithDefinedDictFallback during validation. From that moment on, Pydantic is not involved anymore, and the resulting types are only used for update_from_dict, which is not Pydantic specific (we added it to all Pydantic model in order not to repeat that).